apps/dotcms-ui/src/app/view/components/_common/dot-add-to-menu/dot-add-to-menu.component.ts

```diff
@@ -0,0 +1,164 @@
+
+    constructor(
+        public fb: FormBuilder,
+        private dotMessageService: DotMessageService,
+        private dotMenuService: DotMenuService,
+        private dotAddToMenuService: DotAddToMenuService,
+        private dotNavigationService: DotNavigationService
+    ) {}
+
+    ngOnInit() {
+        this.initForm();
+        this.menu$ = this.dotMenuService.loadMenu().pipe(
+            take(1),
+            tap(() => {
+                this.setDialogConfig(this.form);
+            })
+        );
+    }
+
+    ngOnChanges(changes: SimpleChanges): void {
+        if (changes?.contentType?.currentValue) {
+            this.form.patchValue({
+                title: this.contentType.name
+            });
+            setTimeout(() => {
+                this.titleName.nativeElement.focus();
+            }, 0);
+        }
+    }
+
+    ngOnDestroy(): void {
+        this.destroy$.next(true);
+        this.destroy$.complete();
+    }
+
+    /**
+     * Close dialog modal and reset form
+     * @memberof DotAddToBundleComponent
+     */
+    close(): void {
+        this.cancel.emit(true);
+    }
+
+    /**
+     * Add to bundle if form is valid
+     * @memberof DotAddToBundleComponent
+     */
+    submit(): void {
+        if (this.form.valid) {
+            const cleanPorletId = this.dotAddToMenuService.cleanUpPorletId(
+                this.form.get('title').value
+            );
+            const params: DotCreateCustomTool = {
+                portletId: cleanPorletId,
+                portletName: this.form.get('title').value,
+                contentTypes: this.contentType.variable,
+                dataViewMode: this.form.get('defaultView').value
+            };
+
+            this.dotAddToMenuService
+                .createCustomTool(params)
+                .pipe(take(1), pluck('portlet'))
+                .subscribe(() => {
+                    this.dotAddToMenuService
+                        .addToLayout(`c_${cleanPorletId}`, this.form.get('menuOption').value)
+                        .pipe(take(1), pluck('portlet'))
+                        .subscribe((portletName: string) => {
+                            if (portletName) {
+                                this.dotNavigationService.goTo(`/c/${portletName}`);
+                                this.close();
+                            }
+                        });
+                });
+        }
+    }
+
+    /**
+     * It submits the form from submit button
+     * @memberof PushPublishContentTypesDialogComponent
+     */
+    submitForm(): void {
+        this.formEl.ngSubmit.emit();
+    }
+
+    private initForm(): void {
+        this.form = this.fb.group({
+            defaultView: ['list', [Validators.required]],
+            menuOption: ['', [Validators.required]],
+            title: ['', [Validators.required]]
+        });
+    }
+
+    private setDialogConfig(form: FormGroup): void {
+        this.dialogActions = {
```

Comment:
This could be an Observable and async pipe by combining with the `valueChanges` observable.